### PR TITLE
SB2 Import Modal

### PR DESCRIPTION
**Pull request by kchadha**
### Proposed Changes
This PR is the first step towards adding a user friendly interface for importing Scratch 2.0 projects.

Adds a button in the preview modal to access the new import modal.
Adds a new import modal with input field which expects a link to a scratch project.
Currently this link can be of the following formats:
https://scratch.mit.edu/projects/[project id]/
https://scratch.mit.edu/projects/[project id]
scratch.mit.edu/projects/[project id]/
scratch.mit.edu/projects/[project id]

Anything that does not fall into these formats yields an error state where the input (when focused) has an orange border, and a tooltip (when hovering over the input) with an error message.

Changes to come: more work on the error state, making it closer to design spec (e.g. tooltip should always be visible, and to the right of the "View" button; error message(s) should convey what the error was or why it happened). 
On the vm side, hooking up to scratch-parser, detecting and conveying messaging around projects w/hacked blocks.

### Reason for Changes

Providing a more user friendly interface for importing scratch 2 projects.

### Test Coverage

New import modal gets two integration tests.
Can play around with this work [here](https://llk.github.io/scratch-gui/sb2-import/)

---

- [x] Need to confirm with @carljbowman what should happen when clicking modal header button/"go back" button
- [x] Confirm text in error message
- [x] Get asset from @carljbowman with back arrow
- [ ] Loading state (will follow up with this)

### CODE REVIEW

**Comment by mrjacobbloom** [comment]
Will a hash on the input be supported? The scratch website uses a hash on the URL when switching between the world and the project page, or when jumping to a comment from a notification

File: src/containers/import-modal.jsx [hunk 1]
```diff
@@ -50,9 +50,13 @@ class ImportModal extends React.Component {
         this.setState({inputValue: e.target.value});
     }
     validate (input) {
-        const matches = input.match(/^(https:\/\/)?scratch\.mit\.edu\/projects\/(\d+)(\/?)$/);
-        if (matches && matches.length > 0) {
-            return matches[2];
+        const urlMatches = input.match(/^(https:\/\/)?scratch\.mit\.edu\/projects\/(\d+)(\/?|(\/#editor)?)$/);
```
[the remark this answers is not in the document]
Comment (kchadha) [reply src/containers/import-modal.jsx]: Good catch! It should be fixed now, also added #fullscreen.

**Comment by kchadha** [comment]
@mrjacobbloom Thanks for the catch. I updated to allow #editor, #fullscreen, and #player.

File: src/containers/import-modal.jsx [hunk 2]
```diff
@@ -50,8 +50,7 @@ class ImportModal extends React.Component {
         this.setState({inputValue: e.target.value, hasValidationError: false});
     }
     validate (input) {
-        const urlMatches = input.match(
-            /^(https:\/\/)?scratch\.mit\.edu\/projects\/(\d+)(\/?|(\/#((editor)|(fullscreen)|(player)))?)$/);
+        const urlMatches = input.match(/^(https:\/\/)?scratch\.mit\.edu\/projects\/(\d+)/);
```
Comment (mrjacobbloom) [inline src/containers/import-modal.jsx]: One note here: this isn't too important, but you can use a non-capturing group for the https check with `(?:https:\/\/)` which will clean up the `urlMatches` array a little

**Pull request merged**